network/connection_store.go

```diff
@@ -52,9 +52,11 @@ func (cs *connectionStore) list() []*Connection {
 	cs.rw.RLock()
 	defer cs.rw.RUnlock()
 
-	l := []*Connection{}
+	l := make([]*Connection, len(cs.items))
+	index := 0
 	for _, conn := range cs.items {
-		l = append(l, conn)
+		l[index] = conn
+		index++
 	}
```

Comment:
I find it easier to create an empty slice and the correct capacity and then just us append on it.
This way to code is easy to read and very safe.

Eg:
```
l := make([]*Connection, 0, len(cs.items))
for _, conn := range cs.items {
  l = append(l, conn)
}